pkg/apiserver/handlers.go

```diff
@@ -447,8 +447,13 @@ func WithImpersonation(handler http.Handler, requestContextMapper api.RequestCon
 			actingAsAttributes.Name = name
 		}
 
-		err := a.Authorize(actingAsAttributes)
+		authorized, reason, err := a.Authorize(actingAsAttributes)
 		if err != nil {
+			internalError(w, req)
```

Comment:
pass the error to `internalError` and call runtime.HandleError() on it to make sure we log?